src/main/kotlin/org/jitsi/videobridge/octo/config/OctoConfig.kt

```diff
@@ -16,12 +16,56 @@
 
 package org.jitsi.videobridge.octo.config
 
+import com.typesafe.config.ConfigObject
 import org.jitsi.config.LegacyFallbackConfigProperty
+import org.jitsi.config.legacyConfigAttributes
+import org.jitsi.config.newConfigAttributes
+import org.jitsi.utils.config.FallbackProperty
+import org.jitsi.utils.config.exception.ConfigValueParsingException
+import org.jitsi.videobridge.config.ConditionalProperty
 
 class OctoConfig {
     class Config {
         companion object {
-            class RegionProperty : LegacyFallbackConfigProperty<String>(
+
+            class EnabledProperty : FallbackProperty<Boolean>(
+                // The legacy config file doesn't have an 'enabled' property,
+                // instead it was based on the values of the paremeters.  Here,
+                // we simulate a legacy 'enabled' value based on the results
+                // of validating the other properties in the legacy config
+                // file.
+                legacyConfigAttributes {
+                    name("org.jitsi.videobridge.octo")
+                    readOnce()
+                    retrievedAs<ConfigObject>() convertedBy {
+                        val cfg = it.toConfig()
+                        if (cfg.hasPath("BIND_ADDRESS") && cfg.hasPath("BIND_PORT")) {
+                            val bindAddress = cfg.getString("BIND_ADDRESS")
+                            val bindPort = cfg.getInt("BIND_PORT")
+                            bindAddress != null && (bindPort.isUnpriviligedPort())
+                        } else {
+                            false
+                        }
+                    }
+                },
+                newConfigAttributes {
+                    name("videobridge.octo.enabled")
+                    readOnce()
+                }
+            )
+
+            private val enabledProp = EnabledProperty()
+
+            @JvmStatic
+            fun enabled() = enabledProp.value
+
+            class RegionProperty : ConditionalProperty<String>(
+                ::enabled,
+                Region(),
+                "Octo region is only parsed when Octo is enabled"
```

Comment:
What is the value of `ConditionalProperty` if the getter function will swallow the exception?

Reply:
What do you mean by 'getter function' here, `Region()`?  We try and access the value of that property, which can throw on its own (this doesn't use the `region` getter, but an instance of the property class).